Chameleon Outfit Change (Modified)

Vimas21 opened the pull request:
:cl: Vimas
rcsadd- A cleaned-up format of the chameleon outfit change, this time modified to fit within spookertons code at pull request #28232.
/:cl:

mikomyazaki committed:
Makes chameleon-ness an extension.

SammasC123 committed:
Donezo

Now functions in the new environment- sans headset. I can fix that though.

PsiOmegaDelta commented:
Depends on #28232

SierraKomodo reviewed (commented):
Nothing stands out to me aside from the travis fails - Poking @Spookerton to take a look as it's extending off his existing PR and he better knows if this is using things as intended.

Vimas21 commented:
Travis is giving me a failstate- and I don't see why? Runs fine here, can anyone check?

SierraKomodo commented:
It's getting a lot of chameleon related runtimes. See line 657:

https://travis-ci.com/github/Baystation12/Baystation12/jobs/368829780#L657

PsiOmegaDelta commented:
Noting that this PR will probably have to be rebased once the PR it depends on has been merged, as it does have change requests.

SammasC123 committed:
Bug Quashing

Oh goodness this is in a weird place how did this happen. Either way, fixed the runtime errors

Hubblenaut commented on code/datums/extensions/chameleon.dm:
```diff
@@ -0,0 +1,201 @@
+	var/decl/hierarchy/outfit/selected = outfits[selected_key]
+	var/mob/living/carbon/human/user = usr
+	var/list/equipped = user.get_equipped_items()
+	for (var/obj/item/clothing/clothe in equipped)
```
This loop will currently ignore items like attached accessories and radio headset. Worth pointing out, especially since a lot of Torch jobs use department insignia accessories right now.

Hubblenaut commented on code/datums/extensions/chameleon.dm:
```diff
@@ -0,0 +1,201 @@
+	return sortAssoc(choices)
+
+/atom/proc/change_outfit()
+	set name = "Change Chameleon Outfit"
```
This proc does not have a category set and will therefore not show up in the same category as the other verb. Oversight in latest commit?

Hubblenaut commented on code/datums/extensions/chameleon.dm:
```diff
@@ -0,0 +1,201 @@
+		chameleon_choices = sortAssoc(choices)
+
+	atom_holder = holder
+	chameleon_app += new/atom/proc/chameleon_appearance(atom_holder,"Change [atom_holder.name] Appearance")
```
Noting that since most items have lower case names, this would be at odds with the current naming pattern we have for verbs (upper case nouns).

Hubblenaut commented on code/datums/extensions/chameleon.dm:
```diff
@@ -0,0 +1,201 @@
+/atom/proc/chameleon_appearance()
+	set name = "Change Appearance"
+	set desc = "Activate the holographic appearance changing module."
+	set category = "Object"
```
I realize that the `Object` category was probably not your own choice, but with all the verb cluttering caused by chameleon items (one verb for every differently named chameleon item plus the change outfit verb) it might be a good idea to leave it in category `Chameleon`.

Spookerton commented:
reopen if you intend to revisit this
